kythe/docs/schema/indexing-protobuf.txt

```diff
@@ -31,12 +31,241 @@ implementation, as this is often difficult (or uninteresting) to read. This
 document describes how to encode information about generated code to permit
 cross-language links.
 
+To make the discussion easier to understand let's pretend we are working with
+two languages: SourceLang and TargetLang. SourceLang has `.source` file and TargetLang
+has `.target` files. We also have a tool (generator) that can take generate
+`foo.target` file from `foo.source` file. We have following components:
+
+* Source Indexer - Kythe indexer that takes `.source` files and outputs index
+  data.
+* Target Indexer - Kythe indexer that takes `.target` files and outputs index
+  data.
+* Generator - tool that produces `.target` files from `.source` files.
+* Post processor - Kythe tool that takes all index data produced by all
+  indexers, processes it and outputs final Kythe graph that contains data
+  for both SourceLang and TargetLang.
+
+Now we want to teach Kythe how to create cross-references between generated
+`foo.target` file and original `foo.source` file. The main idea is pretty simple:
+Generator has to output extra data containing mapping of elements in `foo.target`
+to the original elements from `foo.source`. Then when Target Indexer is indexing
+`foo.target` it will use that mapping to output *generates* or *imputes* edges.
+These edges connect nodes from `foo.target` with nodes in `foo.source`.
+
+Kythe doesn't require implementors to use one concrete approach for passing
+mapping metadata and outputting *generates* and *imputes* edges. Below we
+describe two different approaches, each has its own pros and cons. But in
+both cases it is assumed that implementors can change Generator and Target
+Indexer. If possible *generates* approach is preferred as requires less
```

Comment:
Add "the" after "possible" and "it" before "requires"

Reply:
Done.